sandbox/extract-dataset-metadata_emp.R

```diff
@@ -88,4 +89,324 @@ image_count = extract_image_count(exif)
+
+#### white_balance_mode_prop_derived ####
+
+# The proportion of images with a white balance setting that matches the modal white balance setting in the dataset
+
+# BEGIN FUNCTION CODE
+
+exif$white_balance = extract_white_balance(exif)
+
+white_balance_mode_derived = extract_white_balance_mode_derived(exif)
+
+white_balance_mode_prop_derived = (sum(exif$white_balance == white_balance_mode_derived) / nrow(exif))
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_white_balance_mode_prop_derived = function (exif) {
+
+  exif$white_balance = extract_white_balance(exif)
+
+  white_balance_mode_derived = extract_white_balance_mode_derived(exif)
+
+  white_balance_mode_prop_derived = (sum(exif$white_balance == white_balance_mode_derived) / nrow(exif))
+
+  return(white_balance_mode_prop_derived)
+
+}
+
+# test function on exif data
+
+white_balance_mode_prop_derived = extract_white_balance_mode_prop_derived(exif)
+
+white_balance_mode_prop_derived
+
+#### exposure_median_derived ####
+
+# The median exposure time across all images in the dataset. Units: sec
+
+# BEGIN FUNCTION CODE
+
+exposure_median_derived = median(exif$ExposureTime)
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_exposure_median_derived = function (exif) {
+
+  exposure_median_derived = median(exif$ExposureTime)
+
+  return(exposure_median_derived)
+
+}
+
+# test function on exif data
+
+exposure_median_derived = extract_exposure_median_derived(exif)
+
+exposure_median_derived
+
+#### exposure_stdev_derived ####
+
+# The standard deviation of exposure time across all images in the dataset. Units: sec
+
+# BEGIN FUNCTION CODE
+
+exposure_stdev_derived = sd(exif$ExposureTime)
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_exposure_stdev_derived = function (exif) {
+
+  exposure_stdev_derived = sd(exif$ExposureTime)
+
+  return(exposure_stdev_derived)
+
+}
+
+# test function on exif data
+
+exposure_stdev_derived = extract_exposure_stdev_derived(exif)
+
+exposure_stdev_derived
+
+#### image_density ####
+
+# Computed based on image count and footprint. Units: img/ha
+
+# BEGIN FUNCTION CODE
+
+image_density = (nrow(exif))/(units::set_units(sf::st_area(create_mission_polygon(exif, image_merge_distance)), "hectare"))
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_image_density = function (exif, image_merge_distance) {
+
+  image_density = (nrow(exif))/(units::set_units(sf::st_area(create_mission_polygon(exif, image_merge_distance)), "hectare"))
+
+  return(image_density)
+
+}
+
+# test function on exif data
+
+image_density = extract_image_density(exif, image_merge_distance = 50)
+
+image_density
+
+#### area_ha ####
+
+# Area of mission footprint. Units: ha.
+
+# BEGIN FUNCTION CODE
+
+area_ha = units::set_units(sf::st_area(create_mission_polygon(exif, image_merge_distance)), "hectare")
+
+# END FUNCTION CODE
+
+# turn code into a function
+
+extract_area_ha = function (exif, image_merge_distance) {
```

Comment:
I would merge this with the density function to reduce redundancy, and have the new function return a named list of two values.